File: app/routers/events.py
from typing import List

from fastapi import APIRouter, Depends, status
from sqlalchemy.orm import Session

from .. import models, schemas
from ..database import get_db
from ..deps import require_roles
from ..services import event_service, ticket_service

router = APIRouter(prefix="/api/events", tags=["events"])


@router.post("", response_model=schemas.EventResponse, status_code=status.HTTP_201_CREATED)
def create_event(
    data: schemas.EventRequest,
    db: Session = Depends(get_db),
    organizer: models.User = Depends(require_roles(models.Role.ADMIN, models.Role.ORGANIZER)),
):
    return event_service.create_event(db, data, organizer)


@router.get("", response_model=List[schemas.EventResponse])
def get_all_events(db: Session = Depends(get_db)):
    return event_service.get_all_events(db)


@router.get("/upcoming", response_model=List[schemas.EventResponse])
def get_upcoming_events(db: Session = Depends(get_db)):
    return event_service.get_upcoming_events(db)


@router.get("/{event_id}", response_model=schemas.EventResponse)
def get_event(event_id: str, db: Session = Depends(get_db)):
    return event_service.get_event(db, event_id)


@router.get("/{event_id}/tickets", response_model=List[schemas.TicketResponse])
def get_available_tickets(event_id: str, db: Session = Depends(get_db)):
    return ticket_service.get_available_tickets(db, event_id)

File: src/routers/events.js
'use strict';
const express = require('express');
const eventService = require('../services/eventService');
const ticketService = require('../services/ticketService');
const { EventRequest } = require('../schemas');
const { validateBody } = require('../middleware/validate');
const { requireRoles } = require('../middleware/auth');
const { Role } = require('../models/enums');
const { asyncHandler } = require('../utils/errors');

const router = express.Router();

router.post(
  '/',
  ...requireRoles(Role.ADMIN, Role.ORGANIZER),
  validateBody(EventRequest),
  asyncHandler(async (req, res) => {
    const result = eventService.createEvent(req.validated, req.user);
    res.status(201).json(result);
  })
);

router.get(
  '/',
  asyncHandler(async (req, res) => {
    res.json(eventService.getAllEvents());
  })
);

router.get(
  '/upcoming',
  asyncHandler(async (req, res) => {
    res.json(eventService.getUpcomingEvents());
  })
);

router.get(
  '/:eventId',
  asyncHandler(async (req, res) => {
    res.json(eventService.getEvent(req.params.eventId));
  })
);

router.get(
  '/:eventId/tickets',
  asyncHandler(async (req, res) => {
    res.json(ticketService.getAvailableTickets(req.params.eventId));
  })
);

module.exports = router;
